File: frontend/src/types/account.ts
export type AccountStatus = "active" | "paused" | "error" | "archived";
export type Platform = "reddit";

export interface Account {
  id: string;
  nickname: string;
  username: string;
  status: AccountStatus;
  platform: Platform;
  created_at: string;
  updated_at: string;
  last_sync: string | null;
  notes: string | null;
  is_active: boolean;
  session_path: string | null;
  session_status: string | null;
  last_login: string | null;
  last_validation: string | null;
  browser_profile: string | null;
  provider: string | null;
  browser_profile_path: string | null;
  storage_directory: string | null;
  launch_visible_browser: boolean;
  display_name: string | null;
  reddit_username: string | null;
  avatar_url: string | null;
  karma_post: number | null;
  karma_comment: number | null;
  cake_day: string | null;
  verified_email: boolean | null;
  is_nsfw: boolean | null;
  is_moderator: boolean | null;
  is_gold: boolean | null;
  last_profile_sync: string | null;
}

export interface AccountInput {
  nickname: string;
  platform: Platform;
  username: string;
  status: AccountStatus;
  notes?: string | null;
  is_active: boolean;
  provider?: string | null;
  launch_visible_browser: boolean;
}

File: backend/app/models/account.py
from datetime import datetime
from uuid import UUID, uuid4

from sqlalchemy import Boolean, DateTime, Enum, String, Text, UniqueConstraint, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.models.enums import AccountStatus, Platform


class Account(Base):
    __tablename__ = "accounts"
    __table_args__ = (
        UniqueConstraint("platform", "username", name="uq_accounts_platform_username"),
    )

    id: Mapped[UUID] = mapped_column(primary_key=True, default=uuid4)
    nickname: Mapped[str] = mapped_column(String(120), nullable=False)
    username: Mapped[str] = mapped_column(String(120), nullable=False, index=True)
    status: Mapped[AccountStatus] = mapped_column(
        Enum(
            AccountStatus,
            name="account_status",
            values_callable=lambda enum: [item.value for item in enum],
            validate_strings=True,
        ),
        default=AccountStatus.ACTIVE,
        nullable=False,
    )
    platform: Mapped[Platform] = mapped_column(
        Enum(
            Platform,
            name="platform",
            values_callable=lambda enum: [item.value for item in enum],
            validate_strings=True,
        ),
        default=Platform.REDDIT,
        nullable=False,
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )
    last_sync: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    notes: Mapped[str | None] = mapped_column(Text)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    session_path: Mapped[str | None] = mapped_column(String(500))
    session_status: Mapped[str | None] = mapped_column(String(40))
    last_login: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    last_validation: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    browser_profile: Mapped[str | None] = mapped_column(String(160))
    provider: Mapped[str | None] = mapped_column(String(80))
    browser_profile_path: Mapped[str | None] = mapped_column(String(500))
    storage_directory: Mapped[str | None] = mapped_column(String(500))
    saved_username: Mapped[str | None] = mapped_column(String(160))
    saved_password: Mapped[str | None] = mapped_column(String(500))
    remember_credentials: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    auto_login: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    launch_visible_browser: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
